doc/source/operations/secret-rotation.rst

```diff
@@ -0,0 +1,528 @@
+
+          nova-manage cell_v2 update_cell --cell_uuid 00000000-0000-0000-0000-000000000000 --database_connection "CONNECTION WITH NEW PASSWORD HERE" --transport-url "none:///"
+
+       (If the ``cell_uuid`` for cell0 is not
+       ``00000000-0000-0000-0000-000000000000``, change the above command
+       accordingly)
+
+15.  Re-encrypt your ``passwords.yml`` file
+
+16. Stop all OpenStack services
+
+    .. code:: bash
+
+       kayobe playbook run $KAYOBE_CONFIG_PATH/ansible/stop-openstack-services.yml
+
+.. _k-a-change:
+
+17. Delete the service users in Keystone. The exact users will depend on the
+    deployment. Multinode example:
+
+      .. note::
+
+         Alternatively, cherry-pick
+         `this patch <https://review.opendev.org/c/openstack/kolla-ansible/+/903178>`__
+
+
+    .. code:: bash
+
+      openstack user delete glance cinder placement nova neutron heat magnum magnum_trustee_domain_admin barbican designate
+
+18. Flush the Memcached data on all controllers (any old data will now be
+    inaccessible)
+
+    #. Install Telnet (on one of the controllers)
+
+       .. code:: bash
+
+          sudo apt -y install telnet
+
+    #. Check the config for the IP and port used by Memcached (on every
+       controller)
+
+       .. code:: bash
+
+          sudo grep command /etc/kolla/memcached/config.json
+
+       The IP and port will be printed after ``-l`` and ``-p`` respectively
+
+    #. For each controller start a Telnet session, clear all data, then
+       exit
+
+       .. code:: bash
+
+          telnet <ip> <port>
+          flush_all
+          quit
+
+19. Nuke RabbitMQ
+
+    .. code:: bash
+
+       kayobe overcloud host command run -l controllers --become --command "docker stop rabbitmq && docker rm rabbitmq && docker volume rm rabbitmq"
+
+20. Reconfigure Overcloud services to apply changes
+
+      .. warning::
+
+         VMs should continue running, but connections to them will briefly be
+         disrupted when Neutron is redeployed when using ML2/OVS
+
+   .. code:: bash
+
+      kayobe overcloud service deploy
+
+21. Manually update ``heat_domain_admin_password``
+
+    #. TODO: Instructions
+       This has not been tested yet
+
+22. Re-run Tempest to make sure everything has come back
+
+23. Inform other users of the steps they’ll need to take now that the secrets
+    have been rotated:
+
+    1. SSH keys have been rotated, so the new key will have to be distributed
+       if individual user accounts are used
+
+    2. Any existing ``openrc`` files generated by Kolla Ansible will need to be
+       re-generated or edited to use the new Keystone admin password
+
+24. Create a PR to merge the new secrets into your main Kayobe configuration
+    branch
+
+   .. warning::
+
+      Unless you **really** enjoyed this process, RE-ENCRYPT
+      ``passwords.yml`` BEFORE COMMITTING
+
+25. Approximately 1 week after deploying, remove the old horizon secret key
+    from ``passwords.yml`` and reconfigure horizon
```

Comment:
Why approximately, why not right away? Would be nice to have an explanation here.

Reply:
You missed a bit: https://github.com/stackhpc/stackhpc-kayobe-config/pull/832/files#diff-3f806282c2ef8366b9d75791dba5e94f2e7f22be9eaff1b1e7db53b149ae3e25R195-R199